eodag/rest/utils.py

```diff
@@ -454,6 +472,27 @@ def get_criterias_from_metadata_mapping(
     return criterias
 
 
+def _add_to_cache(product_type: str, provider: Optional[str], products: SearchResult):
```

Comment:
provider should be mandatory. We can take it from the first product of SearchResult and assume it is the same for all products of SearchResult

Reply:
provider is still optional here but is added to cache key (provider from product used if not given)